### DISCUSSION TITLE
can i configure red squiggles for links that don't resolve?

### DISCUSSION THREAD

**Discussion by jli** [svsool/vscode-memo#599] [Q&A]
(thanks for memo - it's been great so far!)

I think it would be nice to see when i've made a typo in my links. 

I like how memo makes it easy to create new notes by following links to notes that don't exist. But, I don't like having links that don't resolve, since it might be due to a typo. I feel like having a minimal indicator like a red squiggle would be good.

Is it possible to configure this?

**Comment by svsool** [comment]
Dangling links were highlighted previously before introducing the new syntax injection approach ([this PR](https://github.com/svsool/memo/issues/560)). Maybe @memeplex could confirm whether it's possible or not.

**Comment by memeplex** [comment]
Look into the links I provided for the approach I took in Foam. There we used syntax injection for the base case and decorations for unresolved links. Since unresolved links are a minority, performance and configurability limitations of decorations are less important.

I didn't realize at the time that you were differentiating between resolved and unresolved links, and TBH I still can't see it in the deleted syntaxDecorations.ts, notice that I even pointed out that fact in my proposal:

> Notice that the implementation in foam is harder because they highlight placeholders and actual wikilinks in a different way, but in memo there is no downside

so perhaps I was wrong there. Sorry if that was the case.

**Comment by svsool** [comment]
You are right @memeplex, it was not considered in `syntaxDecorations.ts` even before, I actually realized that dangling links are only highlighted (and were before) with red in read-only mode and not in the main editor window.

<img src="https://user-images.githubusercontent.com/1452141/192832152-8a9738fa-da55-46c3-ba88-c6344522da38.png" width="400" />

I think dual approach is the way to solve this problem as you mentioned.

**Reply by svsool** [reply]
there is btw this pop-up also which reflects state of the link

![image](https://user-images.githubusercontent.com/1452141/193400564-b13b613a-9d70-4aee-9e31-e730c55ca081.png)

**Comment by memeplex** [comment]
OTOH I'm not sure if this is worth the effort. The information that a link is dangling may be seen as:

1. The link is broken
2. This is a topic that I still haven't developed

I guess 1 is not the case of interest here. Regarding 2, if the topic is interesting enough for you to follow the link, you will immediately realize there is work to be done. If the topic is not of interest in the current context, having it highlighted may even be seen as syntax noise. After all, having or not the file is just one of many degrees of development for a topic. Moreover, some maintenance concern like "I want to review all my dangling links" is not optimally addressed from another topic page that I happen to have opened for whatever reasons.

For example, Logseq is a modern note taking system that champions the backlinks = tag approach and all wikilinks, "broken" or not, look the same, because you see them as tags, as concepts that don't necessarily have text of their own, since the linked references / backlinks are always there as part of the concept. So having them syntactically "downgraded" would somehow be against the guiding philosophy.

**Comment by memeplex** [comment]
>> i've made a typo in my links.

> 1. The link is broken
>
> I guess 1 is not the case of interest here

Well, it is after all... Still, considering all my points above, there is a trade off.
